src/main/java/io/aiven/kafka/connect/transforms/ConcatFields.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2019 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.transforms;
+
+import java.util.Map;
+import java.util.StringJoiner;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ConcatFields<R extends ConnectRecord<R>> implements Transformation<R> {
+    private ConcatFieldsConfig config;
+    private static final Logger log = LoggerFactory.getLogger(ConcatFields.class);
+
+    protected abstract String dataPlace();
+
+    protected abstract SchemaAndValue getSchemaAndValue(final R record);
+
+    @Override
+    public ConfigDef config() {
+        return ConcatFieldsConfig.config();
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        this.config = new ConcatFieldsConfig(configs);
+    }
+
+/*
+        final SchemaAndValue schemaAndValue = getSchemaAndValue(record);
+        if (schemaAndValue.schema() == null) {
+            throw new DataException(dataPlace() + " schema can't be null: " + record);
+        }
+
+        final Optional<Object> newValue;
+        if (config.fieldName().isPresent()) {
+            newValue = getNewValueForNamedField(
+                    record.toString(), schemaAndValue.schema(), schemaAndValue.value(), config.fieldName().get());
+        } else {
+            newValue = getNewValueWithoutFieldName(
+                    record.toString(), schemaAndValue.schema(), schemaAndValue.value());
+        }
+
+        if (newValue.isPresent()) {
+            return record.newRecord(
+                    record.topic(),
+                    record.kafkaPartition(),
+                    record.keySchema(),
+                    record.key(),
+                    record.valueSchema(),
+                    newValue.get(),
+                    record.timestamp(),
+                    record.headers()
+            );
+        } else {
+            return record;
+        }
+ */
```

Comment:
Maybe something needs to be done to this leftover

Reply:
haha oops yeah I meant to delete that!